assets/js/modules/analytics/common/account-create.js

```diff
@@ -0,0 +1,131 @@
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../../../components/button';
+import ProgressBar from '../../../components/progress-bar';
+import { trackEvent } from '../../../util';
+import TimezoneSelect from './account-create/timezone-select';
+import AccountField from './account-create/account-field';
+import PropertyField from './account-create/property-field';
+import ProfileField from './account-create/profile-field';
+import CountrySelect from './account-create/country-select';
+import { STORE_NAME, FORM_ACCOUNT_CREATE } from '../datastore/constants';
+import { STORE_NAME as CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
+import { countryCodesByTimezone } from '../util/countries-timezones';
+import Data from 'googlesitekit-data';
+
+const { useDispatch, useSelect } = Data;
+
+export default function AccountCreate() {
+	const accountTicketTermsOfServiceURL = useSelect( ( select ) => select( STORE_NAME ).getAccountTicketTermsOfServiceURL() );
+	const canSubmitAccountCreate = useSelect( ( select ) => select( STORE_NAME ).canSubmitAccountCreate() );
+	const isDoingCreateAccount = useSelect( ( select ) => select( STORE_NAME ).isDoingCreateAccount() );
+	const siteURL = useSelect( ( select ) => select( CORE_SITE ).getReferenceSiteURL() );
+	const siteName = useSelect( ( select ) => select( CORE_SITE ).getSiteName() );
+	const tz = useSelect( ( select ) => select( CORE_SITE ).getTimezone() );
+
+	const [ isNavigating, setIsNavigating ] = useState( false );
+
+	// Redirect if the accountTicketTermsOfServiceURL is set.
+	if ( accountTicketTermsOfServiceURL ) {
+		global.location.assign( accountTicketTermsOfServiceURL );
+	}
+
+	// Set form defaults on initial render.
+	const { setForm } = useDispatch( STORE_NAME );
+	useEffect( () => {
+		const { hostname } = new URL( siteURL );
+		let timezone = tz;
+
+		if ( ! countryCodesByTimezone[ timezone ] ) {
+			timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+		}
+		setForm( FORM_ACCOUNT_CREATE, {
+			accountName: siteName,
+			propertyName: hostname,
+			profileName: __( 'All website traffic', 'google-site-kit' ),
+			countryCode: countryCodesByTimezone[ timezone ],
+			timezone,
```

Comment:
Usually not a fan of ternaries, but in this case it allows us to avoid the two `timezone` variables.
```suggestion
			timezone: ! countryCodesByTimezone[ timezone ] ? Intl.DateTimeFormat().resolvedOptions().timeZone : timezone,
```

(assuming you rename `tz` to `timezone`)

Reply:
refactored to avoid tz usage in https://github.com/google/site-kit-wp/pull/1389/commits/e1deb02800ce9ed14beed8ff4ec4a80656e6547e. We have to actually set timezone to the browser value as well. You can test this by setting WordPress timezone to "UTC" (or any of the UTC+ options). This is not a "real" timezone, so we fall back to the browser timezone.